daemon/containerd/image_pull.go

```diff
@@ -75,6 +75,11 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 }
 
 // GetRepository returns a repository from the registry.
-func (i *ImageService) GetRepository(ctx context.Context, ref reference.Named, authConfig *registry.AuthConfig) (distribution.Repository, error) {
-	return nil, errdefs.NotImplemented(errors.New("not implemented"))
+func (i *ImageService) GetRepository(ctx context.Context, ref reference.Named, authConfig *registry.AuthConfig) (dist.Repository, error) {
+	return distribution.GetRepository(ctx, ref, &distribution.ImagePullConfig{
```

Comment:
I guess one question is; could we implement such services with the containerd distribution code? (should we?)

Reply:
Hmm, I think maybe the reverse would be better - why does it need to be dependent on ImageService?
Could we move it to RegistryService instead?

Reply:
Maybe we could at least move it to `executor.Backend` (implemented by Daemon)? Or would it also involve swarmkit?

Reply:
> Maybe we could at least move it to executor.Backend (implemented by Daemon)? Or does swarmkit/other packages depend on this interface?

Hm.. good one. I should have a look what exactly is passed on to the cluster executor (I do think that interface also expects "push" and "pull", for which we _do_ need the ImageService, so maybe it's not an option currently)